### DISCUSSION TITLE
How to setup the optimal configuration for prisma Postgres running in Docker

### DISCUSSION THREAD

**Discussion by peterHakio** [prisma/prisma#23580] [Q&A]
### Question

I have a data intensive app, where allot of data is fetched when a user opens the main page in around 10 seperate requests. We have an expert system with few active users. We are trying to optimize for parralizabilty and it is okay if there is extra 50 ms in latensy between some connection spinning up. 
Currently are just using a naked:
`export const prisma =  new PrismaClient())`

So I would like to hear how you would improve this. 

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
Postgres 

- Node.js version: 

- prisma                  : 5.11.0

**Comment by ludralph** [comment]
Hello @peterHakio 👋 

Thank you for raising this question.

Can you please tell us more about the application you're building? What database are you using? Is it a serverless application or a long-running application? What stack are you utilizing for your application? How are you deploying your app? Because you want to optimize for parallelization,  how many maximum connections does your database support?

**Comment by peterHakio** [comment]
Hi @ludralph
Sorry I first respond now.
- We are building demand forecasting / inventory planning platform. 
- We are using TRPC, NEXT, Typescript and Prisma
- We are running with docker.
- We are currently using Fly.io, but are soon migrating the Google Cloud, where all the ML is
- The application is a data itensive application, with allot of heavy read opporations
- We only have few users at a time, but every user needs to load multiple megabytes of data
- We are currently have max_connections = 300, but that is not tuned

And thank you very much for returning

**Reply by ludralph** [reply]
Hi @peterHakio 

For your usecase, I'll recommend you to adjust the connection limit in your database connection URL. For your case with heavy read operations and few concurrent users, you might want to start with a higher limit:
```
DATABASE_URL="postgresql://user:password@host:5432/database?connection_limit=50"
```
You should also consider increasing the pool timeout to allow more time for complex queries:
```
DATABASE_URL="postgresql://user:password@host:5432/database?connection_limit=50&pool_timeout=30"
```
Use Prisma's query optimization features like `select` and `include` to fetch only the necessary data. This can significantly reduce the amount of data transferred. You can also leverage [Prisma Optimize](https://www.prisma.io/docs/optimize) to get suggestions and recommendations for improving your query. Ensure your database tables are properly indexed for the queries you're running frequently. This can greatly improve read performance.

For your data-intensive application, you might benefit from using [Prisma Accelerate](https://www.prisma.io/data-platform/accelerate), which provides a globally distributed database cache and connection pooling. These optimizations should help improve the performance of your Prisma setup with PostgreSQL in Docker.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!;
